src/coprocessor/dag/executor/mod.rs

```diff
@@ -237,18 +234,16 @@ impl OriginCols {
     }
 }
 
-pub trait Executor {
+pub trait Executor: Send {
     fn next(&mut self) -> Result<Option<Row>>;
-    fn collect_output_counts(&mut self, counts: &mut Vec<i64>);
-    fn collect_metrics_into(&mut self, metrics: &mut ExecutorMetrics);
-    fn collect_execution_summaries(&mut self, target: &mut [ExecSummary]);
+
+    fn collect_exec_stats(&mut self, dest: &mut ExecuteStats);
+
+    fn collect_storage_stats(&mut self, dest: &mut Statistics);
```

Comment:
If this still depends on `tikv::storage::Statistics`, how can we extract it?
Does this `Executor` trait needs a associated type `Statistics`?

Reply:
Sorry, I didn't saw it in #5119 ...

Reply:
Yes! It will be further isolated by associated types in later PRs. This PR only isolate it from the shared structure.